pontoon/terminology/models.py

```diff
@@ -0,0 +1,250 @@
+                .count()
+
+        )
+        log.info(
+            'Assigned %s terms to %s entities and matched %s phrases.',
+            terms_count,
+            len(entities),
+            phrases_count
+         )
+
+    @transaction.atomic
+    def import_tbx_terms(self, tbx_terms):
+        """
+        Creates/updates term objects passed from terminology files.
+        All terms will be automatically matched with their entities.
+
+        :param VCSTerm tbx_terms: a list of terms imported from the filesystem/vcs.
+        """
+        db_terms = {}
+
+        for term in Term.objects.all():
+            db_terms[term.term_id] = term
+
+        new_terms = []
+        update_terms = []
+        updated_records = 0
+
+        for term_id, term in tbx_terms.items():
+            try:
+                db_term = db_terms[term.term_id]
+                db_term.note = term.note
+                db_term.description = term.description
+                update_terms.append(db_term)
+                updated_records += 1
+            except KeyError:
+                new_terms.append(
+                    Term(
+                        term_id=term.term_id,
+                        source_term=term.source_text,
+                        source_term_singulars=' '.join(get_singulars(term.source_text)),
+                        note=term.note or '',
+                        description=term.description or '',
+                    )
+                )
+
+        if update_terms:
+            bulk_update(update_terms, update_fields=['note', 'description'], batch_size=100)
+            log.info('Updated {} existing terms.'.format(updated_records))
+
+        if new_terms:
+            self.bulk_create(new_terms, 1000)
+            log.info('Inserted {} new terms.'.format(len(new_terms)))
+
+        changed_terms = (new_terms + update_terms)
+        TermTranslation.objects.filter(term__in=changed_terms).delete()
+
+        locale_cache = {locale.code: locale for locale in Locale.objects.all()}
+        update_translations = []
+        for term in changed_terms:
+            vcs_term = tbx_terms[term.term_id]
+            for locale_code, translations in vcs_term.translations.items():
+                for translation in translations:
+                    update_translations.append(
+                        TermTranslation(
+                            locale=locale_cache[locale_code],
+                            text=translation,
+                            term=term,
+                        )
+                    )
+        TermTranslation.objects.bulk_create(update_translations)
+
+        log.info('Updated {} translations of terms.'.format(len(update_translations)))
+        self.assign_terms_to_entities(
+            Entity.objects.values_list(
+                'pk',
+                'string',
+                'string_singulars',
+                'string_plural',
+                'string_plural_singulars'
+            )
+        )
```

Comment:
Everytime we import TBX, `assign_terms_to_entities` is called for all Entities in the DB, removing all existing terms relations and assigning all entities to all terms?